src/services/montage-helper.service.js

```diff
@@ -5,7 +5,27 @@
 		.module('montage')
 		.factory('montageHelper', montageHelper);
 
-	function montageHelper(montage, authService) {
+	function montageHelper(constants, montage, authService, ngProgressFactory) {
+
+		var pendingRequestCount = 0;
+		var progressBar = ngProgressFactory.createInstance();
+		progressBar.setColor(constants.primaryColor);
+
+		var _request = montage.Client.prototype.request;
+
+		montage.Client.prototype.request = function(...args) {
+			pendingRequestCount++;
+			progressBar.start();
+
+			return _request.bind(this)(...args).then(response => {
+				pendingRequestCount--;
+
+				if (!pendingRequestCount) { progressBar.complete(); }
```

Comment:
I was aware that my chained `.then()` function was not returning a value. I guess I should have worded that differently. I'll look into some way of using `.catch()`, in fact I currently have this implemented in a local branch. I'll keep swinging at it.